clis/teliod/src/interface_configurator.rs

```diff
@@ -0,0 +1,120 @@
+use crate::TeliodError;
+use serde::Deserialize;
+use std::net::IpAddr;
+use std::process::Command;
+use tracing::{error, info};
+
+pub trait InterfaceConfigurator {
+    /// Configure the IP address and routes for a given interface
+    fn set_ip(&self, _adapter_name: &str, _ip_address: &IpAddr) -> Result<(), TeliodError> {
+        Ok(())
+    }
+
+    /// Helper function to execute a system command
+    fn execute(&self, command: &mut Command) -> Result<(), TeliodError> {
+        let output = command
+            .output()
+            .map_err(|e| TeliodError::SystemCommandFailed(e.to_string()))?;
+
+        if output.status.success() {
+            Ok(())
+        } else {
+            let stderr = String::from_utf8_lossy(&output.stderr);
+            error!("Error executing command {:?}: {:?}", command, stderr);
+            Err(TeliodError::SystemCommandFailed(stderr.into()))
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum InterfaceConfigurationProvider {
```

Comment:
NIT: might save some boiler plate: https://lib.rs/crates/enum_dispatch

Reply:
I like the idea, but unfortunately it's creating a bit of a conflict with `#[derive(Deserialize)]` that would take me a bit of time to sort out. I added a note to follow up on this.